python/ray/llm/_internal/batch/stages/vllm_engine_stage.py

```diff
@@ -116,14 +119,19 @@ class vLLMEngineWrapper:
 
     def __init__(
         self,
-        *args,
         idx_in_batch_column: str,
         max_pending_requests: int = -1,
         **kwargs,
     ):
         self.request_id = 0
         self.idx_in_batch_column = idx_in_batch_column
         self.task_type = kwargs.get("task", vLLMTaskType.GENERATE)
+        self.model = kwargs.get("model", None)
+        assert self.model is not None
+
+        # LoRA related.
+        self.lora_lock = threading.Lock()
```

Comment:
I think using `threading.Lock()` in not right in the context of asyncio. It will block the event loop. We should actually make `_prepare_request` async imo (later we might add cached weight download to this layer). In this case you should def. use `asyncio.Lock()` instead.

Reply:
Hmm good point. Let me fix it